From 1623342d69c5c75bd74118645dd175bb434ed2fa Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 19 Aug 2019 18:01:40 +0300
Subject: [PATCH] [SW-0] gaudi: add HW workaround for strap pins writing

Seem like bits content at write are not aligned like in read and we need
to do some manipulation (while discarding msb)

Change-Id: I2a173a273791fa288a51765d76df11eb3395952c
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 319331c..5e3924a 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -396,6 +396,13 @@
 	int i;
 	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
 
+	/* H/W bug WA:
+	 * rdata[31:0] = strap_read_val;
+	 * wdata[31:0] = rdata[30:21],1'b0,rdata[20:0]
+	 */
+	boot_strap =	((boot_strap & 0x7FE00000) << 1) |
+			( boot_strap & 0x001FFFFF);
+
 	for (i = 0; i < ARRAY_SIZE(gaudi_pll_cfg); i++) {
 		switch (gaudi_pll_cfg[i].base) {
 		case mmPSOC_CPU_PLL_BASE:
